pageserver/src/walreceiver.rs

```diff
@@ -205,8 +205,8 @@ impl<E: Clone> TaskHandle<E> {
 
         let sender = Arc::clone(&events_sender);
         let handle = tokio::task::spawn(async move {
-            let task_result = task(sender, cancellation_receiver).await;
-            events_sender.send(TaskEvent::End(task_result)).ok();
+            events_sender.send(TaskEvent::Started).ok();
```

Comment:
Noticed that there's no `TaskEvent::Started` posted otherwise, the initial channel value is not propagated.
Also, `async move` is needed here, otherwise the closure won't capture the vars for long enough.

So we have to have this logic in one way or another. I can return the `::End` here too and make this task infallible after all these discoveries, if you think it's more appropriate.